kv/txn_coord_sender.go

```diff
@@ -307,15 +307,22 @@ func (tc *TxnCoordSender) Send(ctx context.Context, call proto.Call) {
 	ctx = tracer.ToCtx(ctx, trace)
 
 	// Process batch specially; otherwise, send via wrapped sender.
-	switch args := call.Args.(type) {
+	switch batchArgs := call.Args.(type) {
```

Comment:
this is also happening here. I've been wrapping in a single-element batch in `(*DistSender).Send()` since the first commit. The reason I'm not doing this here is because it gives you honest multi-element batches, and those will only work if all of the cutting machinery is in place already.
But I think at this point I could rip out a lot of `proto.Request` in lieu of `*proto.BatchRequest`. Part of why I didn't so far is that I didn't want to mess with your change.